apps/website/blog/2023-10-11-frontend-plugin/index.mdx

```diff
@@ -0,0 +1,454 @@
+
+### Prerequisites
+
+To start, please make sure that the following tasks have already been performed:
+
+- [Run Backstage Showcase locally](https://github.com/janus-idp/backstage-showcase/blob/main/showcase-docs/getting-started.md#running-locally-with-a-basic-configuration)
+- [Created the Local System Information backend plugin](https://janus-idp.io/blog/2023/08/23/creating-your-first-backstage-backend-plugin)
+  - We will refer to this plugin as _backend plugin_ for the remainder of this article.
+
+---
+
+## Modifying the Backstage backend
+
+The backend plugin exposes the `/system-info` endpoint that is only available using curl, and is not accessible through the Backstage app by any Frontend plugin. To expose and make use of this backend plugin in our app, we will need to wire the plugin's router into the Backstage backend router.
+
+### Exporting the backend plugin router
+
+- Create a new file named `packages/backend/src/plugins/local-system-info.ts` with the following content:
+
+  ```ts title="packages/backend/src/plugins/local-system-info.ts"
+  import { createRouter } from '@internal/plugin-local-system-info-backend';
+  import { Router } from 'express';
+  import { PluginEnvironment } from '../types';
+
+  export default async function createPlugin(env: PluginEnvironment): Promise<Router> {
+    return await createRouter({
+      logger: env.logger,
+    });
+  }
+  ```
+
+- Wire the backend plugin router into the Backstage backend router by modifying `packages/backend/src/index.ts`:
+
+  - Import the newly added file for our backend plugin into the `packages/backend/src/index.ts`.
+  - Then add the plugin into the backend by adding the following line in the `main` function after the existing `addPlugin` calls.
+
+    ```ts title="packages/backend/src/index.ts" showLineNumbers {3,10}
+      import ...
+      ...
+      import sysInfo from './plugins/local-system-info';
+      ...
+      ...
+      async function main() {
+        ...
+        await addPlugin({ plugin: 'search', apiRouter, createEnv, router: search });
+
+        await addPlugin({plugin: 'sys-info', apiRouter, createEnv, router: sysInfo,});
+        ...
+      }
+    ```
+
+### Modifying the `package.json` to run the backend plugin
+
+The command to start the Backstage (`yarn start`) starts both the app and backend as well as any other workspaces found in the `plugins` directory, which also includes our new backend plugin.
+
+Ideally, one would want to publish the backend and frontend plugins and then simply install that plugin to use it. But since we are setting up both the backend and frontend plugins in our local development environment, the `yarn start` command will try to start the app and the backend plugin along with the Backstage backend. Both the backend plugin as well as Backstage backend processes listen on port `7007`. This will cause a port conflict when the Backstage app is started.
```

Comment:
```suggestion
Ideally, one would want to publish the backend and frontend plugins and then simply install that plugin to use it. But since we are setting up both the backend and frontend plugins in our local development environment, the `yarn start` command will try to start the app and the backend plugin along with the Backstage backend. Both the backend plugin as well as Backstage backend processes listen on port `7007`. This will cause a port conflict when the Backstage app starts.
```